patches/net/minecraft/network/Connection.java.patch

```diff
@@ -16,11 +16,15 @@
      }
  
      @Override
-@@ -135,6 +_,7 @@
+@@ -135,6 +_,11 @@
                      this.disconnect(Component.translatable("disconnect.timeout"));
                  } else {
                      Component component = Component.translatable("disconnect.genericReason", "Internal Exception: " + p_129534_);
-+                    LOGGER.error("Exception caught in connection", p_129534_); // Neo: Always log critical network exceptions
++                    ConnectionProtocol protocol = this.channel.attr(getProtocolKey(this.getSending())).get().protocol();
```

Comment:
Should it maybe be `getReceiving` instead? Not sure

Reply:
I think we're able to just use `this.getInboundProtocol().id()` there instead.

Reply:
The `inboundProtocol` field doesn't exist in 1.20.4, so the only options are the channel attribute or null-checking the `PacketListener` and getting the protocol from that. If you two would prefer using the listener's protocol, then I can change that

Reply:
Listener's protocol is fine, since we there will always be one present when we want to log an error anyway (For the same reason as the `inboundProtocol` would be).